common/lib/chem/chem/chemcalc.py

```diff
@@ -52,110 +53,113 @@ def _orjoin(l):
+        """
+        Remove a tag
+        """
         return n[0]
 
     def ion_suffix(n):
-        '''1. "if" part handles special case
-           2. "else" part is general behaviour '''
-
-        if n[1:][0].node == 'number' and n[1:][0][0][0] == '1':
+        """
+        1. "if" part handles special case
+        2. "else" part is general behaviour
+        """
+        if n[1:][0].label() == 'number' and n[1:][0][0][0] == '1':
             # if suffix is explicitly 1, like ^1-
             # strip 1, leave only sign: ^-
-            return nltk.tree.Tree(n.node, n[2:])
+            return nltk.tree.Tree(n.label(), n[2:])
         else:
-            return nltk.tree.Tree(n.node, n[1:])
+            return nltk.tree.Tree(n.label(), n[1:])
 
     dispatch = {'number': lambda x: nltk.tree.Tree("number", [unparse_number(x)]),
                 'unphased': null_tag,
                 'unsuffixed': null_tag,
                 'number_suffix': lambda x: nltk.tree.Tree('number_suffix', [unparse_number(x[0])]),
                 'suffixed': lambda x: len(x) > 1 and x or x[0],
                 'ion_suffix': ion_suffix,
-                'paren_group_square': lambda x: nltk.tree.Tree(x.node, x[1]),
-                'paren_group_round': lambda x: nltk.tree.Tree(x.node, x[1])}
+                'paren_group_square': lambda x: nltk.tree.Tree(x.label(), x[1]),
+                'paren_group_round': lambda x: nltk.tree.Tree(x.label(), x[1])}
 
     if isinstance(tree, str):
         return tree
 
     old_node = None
-    ## This loop means that if a node is processed, and returns a child,
-    ## the child will be processed.
-    while tree.node in dispatch and tree.node != old_node:
-        old_node = tree.node
-        tree = dispatch[tree.node](tree)
+    # This loop means that if a node is processed, and returns a child,
+    # the child will be processed.
+    while tree.label() in dispatch and tree.label() != old_node:
+        old_node = tree.label()
+        tree = dispatch[tree.label()](tree)
 
     children = []
     for child in tree:
         child = _clean_parse_tree(child)
         children.append(child)
 
-    tree = nltk.tree.Tree(tree.node, children)
+    tree = nltk.tree.Tree(tree.label(), children)
 
     return tree
 
 
 def _merge_children(tree, tags):
-    ''' nltk, by documentation, cannot do arbitrary length
-    groups. Instead of:
-    (group 1 2 3 4)
-    It has to handle this recursively:
-    (group 1 (group 2 (group 3 (group 4))))
+    """
+    nltk, by documentation, cannot do arbitrary length groups.
+    Instead of: (group 1 2 3 4)
+    It has to handle this recursively: (group 1 (group 2 (group 3 (group 4))))
     We do the cleanup of converting from the latter to the former.
-    '''
+    """
     if tree is None:
         # There was a problem--shouldn't have empty trees (NOTE: see this with input e.g. 'H2O(', or 'Xe+').
-        # Haven't grokked the code to tell if this is indeed the right thing to do.
```

Comment:
This code has been doing this for at least 5 years, I have to assume it's doing the right thing, or at least that what it's doing is now expected upstream.